File: cameraintegrator/src/main/java/com/himanshu/cameraintegrator/ImageCallback.java
package com.himanshu.cameraintegrator;


import androidx.annotation.Nullable;

/**
 * Interface that will be used to to deliver results back
 */
public interface ImageCallback {

    /**
     * Invoked for a delivering image results back
     */
    void onResult(@RequestSource.RequestSourceOptions int requestedBy, @Nullable Result result, @Nullable Throwable error);

}

File: cameraintegrator/src/main/java/com/himanshu/cameraintegrator/ImageCallback.kt
package com.himanshu.cameraintegrator

import com.himanshu.cameraintegrator.RequestSource.RequestSourceOptions
import java.io.File

/**
 * Interface that will be used to to deliver results back
 */
interface ImageCallback {
    /**
     * Invoked for a delivering image results back
     */
    fun onResult(@RequestSourceOptions requestedBy: Int, result: Result?, error: Throwable?)
}